test/widgets/compose_box_test.dart

```diff
@@ -405,98 +407,102 @@ void main() {
       matching: find.widgetWithIcon(IconButton, icon));
 
     void checkComposeBoxParts({required bool areShown}) {
-      check(contentFieldFinder().evaluate().length).equals(areShown ? 1 : 0);
+      final inputFieldCount = inputFieldFinder().evaluate().length;
+      areShown ? check(inputFieldCount).isGreaterThan(0) : check(inputFieldCount).equals(0);
       check(attachButtonFinder(Icons.attach_file).evaluate().length).equals(areShown ? 1 : 0);
       check(attachButtonFinder(Icons.image).evaluate().length).equals(areShown ? 1 : 0);
       check(attachButtonFinder(Icons.camera_alt).evaluate().length).equals(areShown ? 1 : 0);
     }
 
-    void checkBanner({required bool isShown}) {
-      final bannerTextFinder = find.text(GlobalLocalizations.zulipLocalizations
-        .errorBannerDeactivatedDmLabel);
-      check(bannerTextFinder.evaluate().length).equals(isShown ? 1 : 0);
+    void checkBannerWithLabel(String label, {required bool isShown}) {
+      check(find.text(label).evaluate().length).equals(isShown ? 1 : 0);
     }
 
-    void checkComposeBox({required bool isShown}) {
+    void checkComposeBoxIsShown(bool isShown, {required String bannerLabel}) {
       checkComposeBoxParts(areShown: isShown);
-      checkBanner(isShown: !isShown);
+      checkBannerWithLabel(bannerLabel, isShown: !isShown);
```

Comment:
This refactor (`checkComposeBox` -> `checkComposeBoxIsShown`) should be fine in commit `compose_box: Replace compose box with a banner when cannot post in a channel`, because that makes it obvious what these changes are used for.